### PR TITLE
Setup circleci config

### LINKED ISSUE #7: Set up CI
## Describe the Feature
We need to set up continuous integration for this repo. I'd like to start with the following checks:

- lint
- flow
- jest

## Possible Implementations
https://github.com/react-native-community/react-native-slider/blob/master/.circleci/config.yml

### PR DESCRIPTION
**Pull request by lukewalczak**
# Overview

Setup circleci config - fixes #7

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    - flow:
+        requires:
+        - linux-checkout
+    - jest:
+        requires:
+        - linux-checkout
+    - android-compile:
+        requires:
+        - linux-checkout
+# Disabled until we have macOS containers enabled
```
Comment (michalchudziak) [inline .circleci/config.yml]: I think macOS is available now, do you mind enabling them?

Comment (lukewalczak) [reply .circleci/config.yml]: Sure

**Comment by matt-oakes** [comment]
Thanks for starting this!

I believe that we're looking to standardise on using the CircleCI Orb to simplify and unify most of the configuration for the `react-native-community` repos. You can read more about it here:

https://github.com/react-native-community/react-native-circleci-orb

An example of how to use it is here:

https://github.com/react-native-community/react-native-netinfo/blob/5fc5379218cdfca7e9eb1dac4a4b517740775161/.circleci/config.yml

I would also recommend following the job pattern from there with a single "test javascript" job, compiling Android on Linux, and compiling and (in the future) testing iOS on MacOS. You should do the `yarn install` inside each job as the architecture can change between jobs and the dependencies might not be compatible.

Don't worry about doing the end-to-end Detox tests in this PR, we can add that in later.

Let me know if you need any pointers 👍

**Comment by lukewalczak** [comment]
@matt-oakes I've rewrited config to use [react-native-community/react-native-circleci-orb](https://github.com/react-native-community/react-native-circleci-orb), but it's failing currently because of missing detox.

**Comment by matt-oakes** [comment]
Thanks for making those changes @lukewalczak! I have made a small change to also disable the publish job until we have that enabled correctly. I'll merge this in now 👍

**Pull request merged**